### PR TITLE
Feature/gh 5584

### PR DESCRIPTION
**Pull request by schuppell**
Needs GH-5547, otherwise mergin is not possible.

Wait until GH-5547 is merged.

### CODE REVIEW

File: projects/core/src/configurator/commons/connectors/configurator-commons.adapter.ts [hunk 1]
```diff
@@ -53,4 +53,15 @@ export abstract class ConfiguratorCommonsAdapter {
...
+  /**
+   * Abstract method to get configuration overview
+   *
+   * @param configId configuration id
+   * @param owner configuration owner
+   */
+  abstract getConfigurationOverview(
+    configId: string,
+    owner: Configurator.Owner
+  ): Observable<Configurator.Configuration>;
```
Comment (ChristophHi) [inline projects/core/src/configurator/commons/connectors/configurator-commons.adapter.ts]: Should return Observable<Configurator.Overview>

File: projects/core/src/configurator/commons/connectors/configurator-commons.connector.ts [hunk 2]
```diff
@@ -54,4 +54,11 @@ export class ConfiguratorCommonsConnector {
   ): Observable<Configurator.Configuration> {
     return this.adapter.readPriceSummary(configuration);
   }
+
+  getConfigurationOverview(
+    configId: string,
+    owner: Configurator.Owner
+  ): Observable<Configurator.Configuration> {
+    return this.adapter.getConfigurationOverview(configId, owner);
```
Comment (ChristophHi) [inline projects/core/src/configurator/commons/connectors/configurator-commons.connector.ts]: Should return Observable<Configurator.Overview>.
Still the effect could instantiate  GetConfigurationOverviewSuccess(configuration) with ownerKey and overview which is then in the reducer added to the store

File: projects/core/src/configurator/commons/connectors/converters.ts [hunk 3]
```diff
@@ -17,3 +17,7 @@ export const CONFIGURATION_PRICE_SUMMARY_NORMALIZER = new InjectionToken<
 export const CONFIGURATION_ADD_TO_CART_SERIALIZER = new InjectionToken<
   Converter<Configurator.AddToCartParameters, any>
 >('ConfigurationAddToCartSerializer');
+
+export const CONFIGURATION_OVERVIEW_NORMALIZER = new InjectionToken<
+  Converter<any, Configurator.Configuration>
```
Comment (ChristophHi) [inline projects/core/src/configurator/commons/connectors/converters.ts]: Also deal with Configurator.Overview

File: projects/core/src/configurator/commons/facade/configurator-commons.service.ts [hunk 4]
```diff
@@ -90,6 +90,18 @@ export class ConfiguratorCommonsService {
       });
   }
 
+  getConfigurationOverview(configuration: Configurator.Configuration) {
```
Comment (ChristophHi) [inline projects/core/src/configurator/commons/facade/configurator-commons.service.ts]: Better 'getConfigurationWithOverview'?

File: projects/core/src/configurator/commons/store/effects/configurator.effect.ts [hunk 5]
```diff
@@ -151,6 +155,30 @@ export class ConfiguratorEffects {
...
+    GetConfigurationOverviewSuccess | GetConfigurationOverviewFail
+  > = this.actions$.pipe(
+    ofType(GET_CONFIGURATION_OVERVIEW),
+    map((action: GetConfigurationOverview) => action.payload),
+    mergeMap(payload => {
+      return this.configuratorCommonsConnector
+        .getConfigurationOverview(payload.configId, payload.owner)
+        .pipe(
+          map((configuration: Configurator.Configuration) => {
+            return new GetConfigurationOverviewSuccess(configuration);
```
Comment (ChristophHi) [inline projects/core/src/configurator/commons/store/effects/configurator.effect.ts]: Constructor of OverviewSucces changed to (owner, overview)?

File: projects/storefrontlib/src/cms-components/configurator/commons/config-overview-attribute/config-overview-attribute.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        },
+      })
+      .compileComponents();
+  }));
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ConfigOverviewAttributeComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create component', () => {
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/configurator/commons/config-overview-attribute/config-overview-attribute.component.spec.ts]: Should we test that one can pass values to this component?

File: projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.html [hunk 7]
```diff
@@ -1,3 +1,28 @@
...
+      </section>
+    </ng-container>
+  </ng-container>
 </ng-container>
+
+<ng-template #noAttributes>
+  <section class="cx-config-overview-no-attributes">
+    <!-- TODO: Change text, as we currently do not support filtering on overview page -->
+    <h2>No Results</h2>
+    <p>Remove filter(s) to see Overview content</p>
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.html]: Translate text...

File: projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.ts [hunk 8]
```diff
@@ -26,9 +26,31 @@ export class ConfigOverviewFormComponent implements OnInit {
     this.configuration$ = this.configRouterExtractorService
       .extractConfigurationOwner(this.routingService)
       .pipe(
-        switchMap(owner =>
+        mergeMap(owner =>
```
[the remark this answers is not in the document]
Comment (schuppell) [reply projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.ts]: Done

File: projects/core/src/model/configurator.model.ts [hunk 9]
```diff
@@ -53,6 +54,21 @@ export namespace Configurator {
...
   }
 
+  export interface Overview {
+    groups?: GroupOverview[];
+  }
+
+  export interface GroupOverview {
+    id: string;
+    groupDescription: string;
+    characteristicValues: CharacteristicOverview[];
```
[the remark this answers is not in the document]
Comment (schuppell) [reply projects/core/src/model/configurator.model.ts]: Done.

File: projects/core/src/model/configurator.model.ts [hunk 10]
```diff
@@ -53,6 +54,21 @@ export namespace Configurator {
...
+  }
+
+  export interface CharacteristicOverview {
+    characteristic: string;
```
[the remark this answers is not in the document]
Comment (schuppell) [reply projects/core/src/model/configurator.model.ts]: Done.

File: projects/core/src/configurator/commons/connectors/configurator-commons.connector.ts (hunk 2 above)
Comment (schuppell) [reply projects/core/src/configurator/commons/connectors/configurator-commons.connector.ts]: Done.

File: projects/core/src/configurator/commons/connectors/converters.ts (hunk 3 above)
Comment (schuppell) [reply projects/core/src/configurator/commons/connectors/converters.ts]: Done.

File: projects/core/src/configurator/commons/store/effects/configurator.effect.ts (hunk 5 above)
Comment (schuppell) [reply projects/core/src/configurator/commons/store/effects/configurator.effect.ts]: Done.

File: projects/core/src/configurator/commons/connectors/configurator-commons.adapter.ts (hunk 1 above)
Comment (schuppell) [reply projects/core/src/configurator/commons/connectors/configurator-commons.adapter.ts]: Done.

File: projects/core/src/configurator/commons/facade/configurator-commons.service.ts (hunk 4 above)
Comment (schuppell) [reply projects/core/src/configurator/commons/facade/configurator-commons.service.ts]: Done.

File: projects/storefrontlib/src/cms-components/configurator/commons/config-overview-attribute/config-overview-attribute.component.spec.ts (hunk 6 above)
Comment (schuppell) [reply projects/storefrontlib/src/cms-components/configurator/commons/config-overview-attribute/config-overview-attribute.component.spec.ts]: Done.

File: projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.html (hunk 7 above)
Comment (schuppell) [reply projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.html]: Done.

File: projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.ts [hunk 11]
```diff
@@ -28,7 +28,29 @@ export class ConfigOverviewFormComponent implements OnInit {
...
+
+  hasAttributes(configuration: Configurator.Configuration): boolean {
+    if (
+      configuration.overview === undefined ||
+      configuration.overview === null
+    ) {
+      return false;
+    }
+
+    for (let g = 0; g < configuration.overview.groups.length; g++) {
```
Comment (ChristophHi) [inline projects/storefrontlib/src/cms-components/configurator/commons/config-overview-form/config-overview-form.component.ts]: We should do this in functional style:     return (
      configuration.overview.groups.filter(group => group.attributes.length > 0)
        .length > 0

**Pull request merged**